src/textual/widgets/_directory_tree.py

```diff
@@ -203,25 +203,29 @@ def render_label(
 
         if node._allow_expand:
             prefix = ("📂 " if node.is_expanded else "📁 ", base_style + TOGGLE_STYLE)
-            node_label.stylize_before(
-                self.get_component_rich_style("directory-tree--folder", partial=True)
-            )
+            if self.app.is_mounted(self):
```

Comment:
I have a feeling this would be better done before `render_label`.

It doesn't make much sense for `render_label` to be called prior to mount. Does the user even see the label rendered pre-mount?

Can we bailed out earlier than this?

Reply:
The things is, setting show_root prematurely invokes this code, but it will run again post mount. The change here makes the error go away but I think the issue is that this code runs at all.

It's also an annoying precedent, in that we may get the same issue with other renderables and watches. And I don't want to establish this as a pattern.

Can you investigate a solution, which doesn't place the burden on the widget developer?

Reply:
Do you mean investigate a different approach to how Tree works, or how watches work?

Reply:
Honestly, could be either. Whatever is the most elegant solution.